src/app/services/send-page-data/send-page-data.ts

```diff
@@ -0,0 +1,211 @@
+        'invalidTransferAmount': false,
+        'transferTooSmall': false,
+        'transferTooLarge': false,
+        'navToNavTransfer': false,
+        'changellyError': false,
+      }
+    }
+    this.dataSubject.next(this.dataBundle)
+  }
+
+  checkIsDataSet():boolean {
+    return this.isDataSet
+  }
+
+  storeData(transferAmount, originCoin, destCoin, destAddr): void {
+    this.dataSetSubject.next(false)
+    this.resetDataBundleErrors(this.dataBundle)
+    this.dataStored = false
+    this.dataBundle.transferAmount = Number(transferAmount) ? Number(transferAmount): undefined
+    this.dataBundle.originCoin = originCoin
+    this.dataBundle.destCoin = destCoin
+    this.dataBundle.destAddr = destAddr
+    this.dataStored = true
+    if(!this.validateFormData(this.dataBundle)) {
+      this.dataSubject.next(this.dataBundle)
+      return //validation errors, so return early
+    }
+
+    this.getEstimatedExchange(originCoin, 'nav', transferAmount)
+      .then((data) => {
+        if(originCoin === 'nav'){
+          this.dataBundle.changellyFeeTotalToNav = 0
+        } else {
+          this.dataBundle.changellyFeeTotalToNav = transferAmount - transferAmount * this.CHANGELLY_FEE
+        }
+        this.dataBundle.estConvToNav = this.dataBundle.changellyFeeTotalToNav
+        this.dataBundle.navtechFeeTotal = this.dataBundle.estConvToNav - this.dataBundle.estConvToNav * this.NAVTECH_FEE
+        this.dataBundle.estNavAfterNavtechFee = this.dataBundle.estConvToNav - this.dataBundle.changellyFeeTotalToNav
+
+        this.getEstimatedExchange('nav', destCoin, this.dataBundle.estNavAfterNavtechFee)
+        .then((data) => {
+          if(destCoin === 'nav'){
+            this.dataBundle.changellyFeeTotalFromNav = 0
+          } else {
+            this.dataBundle.changellyFeeTotalFromNav = this.dataBundle.estNavAfterNavtechFee - this.dataBundle.estNavAfterNavtechFee * this.CHANGELLY_FEE
+          }
+          this.dataBundle.estConvFromNav = data
+          this.isDataSet = true
+
+          this.validateDataBundle(this.dataBundle)
+          .then(() => this.dataSubject.next(this.dataBundle))
+        })
+    })
+  }
+
+  validateFormData(dataBundle):boolean {
+    let validData = true
+    if(!Number.isInteger(dataBundle.transferAmount)){
+      dataBundle.errors.invalidTransferAmount = true
+      dataBundle.validData = false
+      validData = false
+    }
+    if(dataBundle.originCoin === 'nav' && dataBundle.destCoin === 'nav') {
+      dataBundle.errors.navToNavTransfer = true
+      dataBundle.validData = false
+      validData = false
+    }
+    if(dataBundle.transferAmount < this.getMinTransferAmount(dataBundle.originCoin, 'nav')) {
+      dataBundle.errors.transferTooSmall = true
+      dataBundle.validData = false
+      validData = false
+    }
+    return validData
+  }
+
+  validateDataBundle(dataBundle) {
+    return new Promise<any>( resolve => {
+      if(dataBundle.estNavAfterNavtechFee > this.MAX_NAV_PER_TRADE) {
```

Comment:
10k NAV is the limit including the NavTech fee. so i dont know what this variable holds, but it sounds like its after the 0.5% is taken away which could result in a tx still too large for NavTech to handle.